### PR TITLE
Sample Issue Tests against YB Versions

### PR DESCRIPTION
**Pull request by makalaaneesh**
### Describe the changes in this pull request
Adding unit tests for issues.

Workflow that will help us detect when issues are fixed in YB:
1. Whenever we introduce a new issue, write a unit test asserting that it fails.
2. The unit test will be run against all the latest YB versions in every series that we support (2.18,2.20,2.23,2024.1,2024.2 for example)
3. YB fixes an issue in 2025.1.0.0. 
4. As part of qualifying voyager for 2025.1 release, all unit tests are also run against 2025.1. Since the issue was fixed, "assert that it fails" will fail. 
5. This prompts us to a. Add `MinimumVersionsFixedIn[2025.1] = 2025.1.0.0` to the issue. b. Write a unit test that asserts that  `parserDetector.GetAllIssues(..., 2025.1)` does not return an issue of that type. 
6. The failing unit test will now start passing because it takes into account `MinimumVersionsFixedIn[2025.1]`. (this part is not implemented yet)

### Describe if there are any user-facing changes
N/A

### How was this pull request tested?
N/A. Adding unit tests.

### Does your PR have changes that can cause upgrade issues? 
| Component        | Breaking changes? |
| :----------------------------------------------: | :-----------: |
| MetaDB                                                    |  No |
| Name registry json                                        |  No |
| Data File Descriptor Json                                 |  No |
| Export Snapshot Status Json                               | No |
| Import Data State                                         | No |
| Export Status Json                                        | No |
| Data .sql files of tables                                 | No |
| Export and import data queue                              | No |
| Schema Dump                                               | No |
| AssessmentDB                                              | No |
| Sizing DB                                                 | No |
| Migration Assessment Report Json                          | No |
| Callhome Json                                             | No |
| YugabyteD Tables                                          | No |
| TargetDB Metadata Tables                                  | No |

### CODE REVIEW

File: yb-voyager/src/queryissue/ddl_issues_test.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	success = t.Run(fmt.Sprintf("%s-%s", "xml functions", ybVersion), testXMLFunctionIssue)
+	assert.True(t, success)
+
+	success = t.Run(fmt.Sprintf("%s-%s", "stored generated functions", ybVersion), testStoredGeneratedFunctionsIssue)
+	assert.True(t, success)
+
+	success = t.Run(fmt.Sprintf("%s-%s", "unlogged table", ybVersion), testUnloggedTableIssue)
+	assert.True(t, success)
+
+	yugabytedbContainer.Terminate(context.Background())
```
Comment (sanyamsinghal) [inline yb-voyager/src/queryissue/ddl_issues_test.go:110]: you already have defer terminate

**Review by sanyamsinghal — COMMENTED** [review]
There are two approaches to figure the supporting versions for an issue:
1. For every issue check the support on all version till date.
2. For every issue check the support on the latest version, if it is supported then we can look at some other versions(or last few series)

Is that correct? 

Our code can have some stale issues also, which be fixed long time back or not very recently.

So how do we plan to ensure that for an issue if it works on latest, then what was the version yugabytedb started supporting it.

**Comment by makalaaneesh** [comment]
> Our code can have some stale issues also, which be fixed long time back or not very recently.
So how do we plan to ensure that for an issue if it works on latest, then what was the version yugabytedb started supporting it.

(Not sure I understand your question completely, we can discuss on call)
These unit tests will only help in detecting cases where issues are fixed **going forward**. For the ones that are already supported, we will have to do the manual work of figuring out which version was support added in . (Although I don't think there are a lot of cases)

File: .github/workflows/issue-tests.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        # using s3 release instead of docker image to allow testing against un-released versions
+        wget https://s3.us-west-2.amazonaws.com/releases.yugabyte.com/${YB_VERSION}/yugabyte-${YB_VERSION}-centos-x86_64.tar.gz
+        mkdir -p yugabyte-${YB_VERSION}
+        tar -xvzf yugabyte-${YB_VERSION}-centos-x86_64.tar.gz -C yugabyte-${YB_VERSION} --strip-components=1
+        yugabyte-${YB_VERSION}/bin/yugabyted start --advertise_address 127.0.0.1
+        sleep 20
+    
+    - name: Test YugabyteDB connection
+      run: |
+        psql "postgresql://yugabyte:@127.0.0.1:5433/yugabyte" -c "SELECT version();"
```
Comment (priyanshi-yb) [inline .github/workflows/issue-tests.yml:39]: nit: `psql $YB_CONN_STR -c "SELECT version();"`

**Review by priyanshi-yb — APPROVED** [review]
LGTM

**Pull request merged**